[BUG] Dotter deploy unable to run after the first time on macos due to no /proc/self

## Environment
- **OS**: macOS Big Sur
- **Dotter version**: Dotter 0.12.3

## Description
Dotter deploy works the first time I run it, but errors on subsequent attempts to run the command.

## Reproduction
A minimal reproducible example: 
1. Create `.dotter/global.toml`:
  ```toml
  [bash.files]
  bashrc = "~/.bashrc"
  ```
2. And `.dotter/local.toml`:
  ```toml
  packages = ["bash"]
  ```
3. Run `touch bashrc` to create the `bashrc` file.
3. Run `dotter` to deploy `bashrc`.
4. Immediately run `dotter` again.

### Expected behavior
The first time `dotter` is run, it should successfully deploy the file (which it does). The second time `dotter` is run, it should successfully deploy again (although there will be no change to the target).

### Actual behavior
The second time I run `dotter`, I get this error:
```
[ERROR] Failed to update symlink "bashrc" -> "/Users/my_username/.bashrc"
Caused by:
    set target symlink owner
    detect if file is owned by the current user
    get metadata of /proc/self
    No such file or directory (os error 2)
```
Running with `RUST_LOG=trace dotter` doesn't give me any other output.

I'm not an expert on any of this, but I guess this is happening because macos doesn't have `/proc/self` like linux does. The error comes from this [line](https://github.com/SuperCuber/dotter/blob/a6ee8f17330796b55db14a1a372c953f469c6054/src/filesystem.rs#L269-L272). 

I'm a beginner in rust, but I would also be happy to try to contribute a bugfix.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,3 +34,6 @@ mockall = "0.9.*"
 
 [target.'cfg(windows)'.dependencies]
 dunce = "1.*"
+
+[target.'cfg(unix)'.dependencies]
+libc = "0.2.94"

--- a/src/filesystem.rs
+++ b/src/filesystem.rs
@@ -266,10 +266,7 @@ impl RealFilesystem {
     fn is_owned_by_user(&self, path: &Path) -> Result<bool> {
         use std::os::unix::fs::MetadataExt;
         let file_uid = path.metadata().context("get file metadata")?.uid();
-        let process_uid = std::path::PathBuf::from("/proc/self")
-            .metadata()
-            .context("get metadata of /proc/self")?
-            .uid();
+        let process_uid = unsafe { libc::geteuid() };
         Ok(file_uid == process_uid)
     }
 }

